specs/file_hierarchy_for_the_verification_of_os_artifacts.md

```diff
@@ -0,0 +1,615 @@
+
+## Terminology
+
+This text uses the generic term _"signature verifier"_ for cryptographic objects that are used to verify signatures.
+In practice, those may be for example X.509 or OpenPGP certificates (the latter are also referred to as public keys).
+
+All technologies for validation of signatures rely on _public key material_.
+However, some systems use bare _public key material_, while others combine key material with additional metadata, forming composite objects that may e.g. also make identity assertions.
+
+This text uses the term _signature verifier_ to refer to two types of verifiers:
+
+- _artifact verifiers_, which are used for the validation of signatures on artifacts,
+- _[trust anchor]s_, which are used to ascertain the authenticity of _artifact verifiers_.
+
+### Revocation of verifiers
+
+Some technologies have a concept of invalidation of verifiers, using a mechanism that signals to users that a verifier should not be relied on anymore (e.g. because the private key material has been compromised).
+
+### Classification of signature verification models
+
+For the use in the VOA structure, digitial signature technologies may be regarded outside of their usual context.
+To simplify and allow a unified view on these technologies, VOA classifies these three distinct families:
+
+#### Point to Point
+
+(e.g. SSH, minisign, signify)
+
+Keys are "atomic" and don't have formal relationships with each other: A key issues a number of signatures, which can be verified.
+
+These systems use "set-style trust": Users choose to rely on some _set_ of keys/verifiers, and are willing to accept signatures made with those.
+
+This model doesn't require or support the use of _trust anchors_ in VOA.
+In this model, the set of _artifact verifiers_ is considered implicitly trusted and used directly for the verification of artifacts.
+
+#### Decentralized delegation
+
+(e.g. OpenPGP)
+
+In these systems, key material and certifications (as used in the OpenPGP "Web of Trust") are decentralized, forming a non-hierarchical network.
+Users need to specify the _trust anchors_ they want to rely on, within their context.
+
+Two approaches are possible for applications using VOA:
+
+- Directly rely on _artifact verifiers_ for a [purpose].
+- Rely on _trust anchors_ which delegate to one or more _artifact verifiers_.
+  Anchors must be explicitly defined in the relevant "trust-anchor" VOA directory (see [purpose]).
+
+#### Hierarchical delegation
+
+(e.g. X.509, SSH+CA)
+
+Other systems have an inherent assumption of hierarchical delegation, formalized by cryptographic certifications.
+In such systems, globally or locally accepted _trust anchors_ are used as starting points in signature validation.
+Certificates and the delegation relations between them form a tree structure.
```

Comment:
This is a bit of a nitpick, but this is not a tree, but more an acyclic graph (and even then, maybe the acyclic is wrong?)

Reply:
We have added a clarification.

Do note, that for the purpose of this document we are sometimes referring to subsets of functionality of certain technologies (to make a point about how this technology is used in the context of VOA).

In case you would still like to see the text improved, don't hesitate to reopen!